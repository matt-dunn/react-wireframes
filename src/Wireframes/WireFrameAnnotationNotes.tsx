/** !
 * Copyright (c) 2019, Matt Dunn
 *
 * @author Matt Dunn
 */

import React from "react";
import styled from "@emotion/styled";
import { IdentifierBase } from "./styles";

import { WireFrameComponent, WireFrameComponents } from "./api";

type WireFrameAnnotationsNotesProps = {
    components: WireFrameComponents;
    highlightedNote?: WireFrameComponent;
}

const WireFrameAnnotationsNotesContainer = styled.ul`
  overflow: auto;
  font-size: 0.85rem;
  list-style: none;
  margin: 0;
  padding: 0;
  
  li {
    padding: 6px 0;
    border-bottom: 1px solid #ccc;
    
    &:last-child {
      border-bottom: none;
    }
    
    &.highlight {
      background-color: rgba(64, 134, 247, 0.25);
    }
    
    header {
      display: flex;
      padding: 0 10px;
      margin-bottom: 5px;
      align-items: center;
      
      h2 {
        font-size: 1.2rem;
        margin: 0;
        font-weight: normal;
      }
    }
    
    article {
      padding: 0 10px;
    }
  }
`;

const IdentifierNote = styled(IdentifierBase)`
  margin-right: 0.5em;
  border-width: 1px;
  border-style: solid;
  font-size: 0.8em;
`;

export const WireFrameAnnotationsNotes = ({ components, highlightedNote }: WireFrameAnnotationsNotesProps) => (
  <WireFrameAnnotationsNotesContainer id="wf-annotations">
    {components.map(component => (
      <li
        key={component.id}
        id={`wf-annotation-${component.id}`}
        className={(highlightedNote === component && "highlight") || ""}
      >
        <header>
          <IdentifierNote>
            {component.id}
          </IdentifierNote>
          <h2>
            {component.options.title}
          </h2>
        </header>
        <article>
          {component.options.description}
        </article>
      </li>
    ))}
  </WireFrameAnnotationsNotesContainer>
);
