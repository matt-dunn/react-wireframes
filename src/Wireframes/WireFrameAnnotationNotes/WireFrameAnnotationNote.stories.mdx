import { Meta, Story, Props, Preview } from '@storybook/addon-docs/blocks';
import { withKnobs, text, boolean, number } from "@storybook/addon-knobs";

import { WireFrameAnnotationsNote } from "./WireFrameAnnotationNote";
import { annotationLongDescription } from "./WireFrameAnnotationNotes.stories.fixtures";

<Meta title="@matt-dunn/react-wireframes/WireFrameAnnotationsNote" component={WireFrameAnnotationsNote} decorators={[withKnobs]} />

# WireFrameAnnotationsNote

Lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi ornare ligula nec leo facilisis ultricies. Fusce quis massa finibus neque volutpat commodo. Nunc blandit dolor in mi viverra, sit amet imperdiet lectus euismod. Mauris et urna vitae dolor accumsan vehicula quis ut urna. Aenean tincidunt enim in pellentesque fringilla. Duis sagittis sem ac quam egestas elementum. Nulla a velit a diam congue euismod.

<Props of={WireFrameAnnotationsNote} />

Pellentesque non sapien a dolor aliquet tempor sit amet non nunc. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Suspendisse tristique nulla vitae vestibulum lacinia. Cras sit amet luctus metus. Nunc malesuada erat eu magna dignissim porta. Sed vitae porttitor ligula. Fusce velit est, dapibus vehicula odio a, bibendum ornare nisi. Aenean sed odio et ipsum laoreet facilisis. Proin et lorem eget elit pellentesque tincidunt. Nullam ac urna sed massa tincidunt accumsan. Ut ultrices nibh ut consectetur tincidunt. Donec eget ullamcorper eros, vel congue mi. Suspendisse sapien erat, sagittis eget cursus nec, aliquam in mi.

<Preview>
    <Story name="Annotation">
        <WireFrameAnnotationsNote
            annotation={{
                         id: number("id", 1),
                         count: 1,
                         Component: () => {},
                         options: {
                           title: text("Title", "Test component"),
                           description: text("Description", "Test description"),
                         },
                       }}
        />
    </Story>
    <Story name="Annotation with long description">
        <WireFrameAnnotationsNote
            annotation={annotationLongDescription}
        />
    </Story>
</Preview>

