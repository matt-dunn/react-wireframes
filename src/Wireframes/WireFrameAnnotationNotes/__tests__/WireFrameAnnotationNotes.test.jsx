/** !
 * Copyright (c) 2019, Matt Dunn
 *
 * @author Matt Dunn
 */

import React from "react";
import { shallow } from "enzyme";

import { WireFrameAnnotationsNotes as Component } from "../WireFrameAnnotationNotes";

describe("Wireframe: WireFrameAnnotationsNotes", () => {
  let MockedComponent;
  let annotations;

  beforeEach(() => {
    MockedComponent = jest.fn();

    annotations = [
      {
        id: 1,
        count: 1,
        Component: MockedComponent,
        options: {
          title: "Test component",
          description: "Test description",
        },
      },
    ];
  });

  it("should render correctly", () => {
    const highlightedNote = undefined;

    const wrapper = shallow(
      <Component
        annotations={annotations}
        highlightedNote={highlightedNote}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("should render correctly with highlight", () => {
    const highlightedNote = annotations[0];

    const wrapper = shallow(
      <Component
        annotations={annotations}
        highlightedNote={highlightedNote}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
