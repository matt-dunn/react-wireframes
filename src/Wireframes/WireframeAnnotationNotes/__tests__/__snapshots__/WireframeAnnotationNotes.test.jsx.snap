// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wireframe: WireframeAnnotationsNotes should render correctly 1`] = `
<WireframeAnnotationNotesContainer
  tabIndex={0}
>
  <li
    data-annotation-id={1}
    data-highlighted={false}
    key="1"
  >
    <WireframeAnnotationNote
      annotation={
        Object {
          "Component": [MockFunction],
          "count": 1,
          "id": 1,
          "options": Object {
            "description": "Test description",
            "title": "Test component",
          },
        }
      }
      isHighlighted={false}
    />
  </li>
</WireframeAnnotationNotesContainer>
`;

exports[`Wireframe: WireframeAnnotationsNotes should render correctly with highlight 1`] = `
<WireframeAnnotationNotesContainer
  tabIndex={0}
>
  <li
    data-annotation-id={1}
    data-highlighted={true}
    key="1"
  >
    <WireframeAnnotationNote
      annotation={
        Object {
          "Component": [MockFunction],
          "count": 1,
          "id": 1,
          "options": Object {
            "description": "Test description",
            "title": "Test component",
          },
        }
      }
      isHighlighted={true}
    />
  </li>
</WireframeAnnotationNotesContainer>
`;
